README.md

```diff
@@ -31,8 +45,161 @@ $ docker-compose up
+### Setting up the development environment
 
-- Add ability to sort key listing lexicographically in ascending or descending order.
-- Add ability to filter keys by some prefix query.
-- Add ability to delete keys.
+Create a development environment that is conducive for the candidate to edit
+and exercise their code. Candidates will likely have specific environments
+they prefer to work in, but for efficiency our environment will have common
+general-purpose tools. Ensure these are all in working order, and that the
+server is running before the interview starts.
+
+1. [Visual Studio Code](https://code.visualstudio.com/download) with Python
+   and Javascript syntax highlighting
+2. [Postman](https://www.getpostman.com/downloads/) or other graphical HTTP
+   request tool
+3. Web browser and internet connection
+4. Command line terminal
+5. Running server, in the language preferred by the candidate
+
+## The interview
+
+### Baseline starting implementation
+
+The baseline server implementation has 3 endpoints:
+
+- `/`: serves a welcome message and confirms the site is running
+- `/set/?{key}={value}`: persists the key and value to an in-memory data structure
+- `/keys/`: returns all keys currently persisted
+
+The server does not have any explicit bugs or "gotchas" that are intended to trip
+up or confuse the candidate. Instead, it is left with intentional shortcomings and
+areas for enhancements. Notably:
+
+- The data is stored in-memory. The data is not persisted when auto-reloading the
+  server after source files change.
+- There is no explicit error handling.
+- No ability to fetch individual keys.
+- Not all methods restrict the HTTP verb as appropriate (`/set` uses `GET`).
+
+The interview process should be structured around making incremental improvements
+to the server and evaluating the candidate's ability to understand the
+codebase and requirements, and perform the actual implementation. Prefer to
+guide the exercise in a way that gives you insight into the candidate's
+skills and behaviors. For example, if the candidate is getting
+stuck and is unable to use your prompts to make progress, note it and move on to
+another feature requirement.
+
+The candidate is allowed to use whatever resources would be available to them during
+the normal course of their job. For example, looking up documentation or
+using Stack Overflow is acceptable, as well as asking questions to the
+interviewer. How they go about these are all valuable signals for the
+interview.
+
+### Programming tasks
+
+#### 1. General understanding
+
+Before the first task, ask the candidate to evaluate the code in the main
+server source code file. Invite them to ask you questions or otherwise
+clarify any confusing elements. Allow them to scan the framework documentation
+or explore the Docker file that creates the server if they express an interest.
+
+##### Evaluation criteria
+
+1. The candidate may not be proficient in either programming language
```

Comment:
This feels less like evaluation criteria and more a condition to help the reviewer navigate _if` the candidate is unfamiliar with either language. I'd consider revising this to:

> Are they able to familiarize themselves with a foreign codebase effectively?